### PR TITLE
Redux store

### PR DESCRIPTION
**Pull request by johncalesp**
- State management is handle by redux-store
- Changed unit testing

### CODE REVIEW

**Comment by michaelshin** [comment]
Can you add more details on what you moved to Redux and what each reducer is responsible for?

File: deepview-explore/react-ui/src/redux/slices/vsCodeSlice.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+// https://stackoverflow.com/questions/54135313/webview-extension-in-typescript
+/**
+ * Returns the vscode API handle. the acquireVsCodeApi function would not be available when compiling
+ * the react project, so we need to dynamically look it up. This function also caches the handle and
+ * returns it if previously acquired.
+ * @returns The VSCode API handle
+ */
+function acquireApi() {
+  // if (typeof this.acquireApi.api == 'undefined') {
```
Comment (michaelshin) [inline deepview-explore/react-ui/src/redux/slices/vsCodeSlice.js:11]: remove this comment

**Comment by michaelshin** [comment]
I think this is a good start for us to use Redux. Did you have a way to separate what states should be in the store vs stored locally? If we can keep that consistent, then it would be better to maintain longterm

File: deepview-explore/react-ui/src/redux/slices/vsCodeSlice.js (hunk 1 above)
Comment (johncalesp) [reply deepview-explore/react-ui/src/redux/slices/vsCodeSlice.js]: got it

File: deepview-explore/react-ui/src/sections/Habitat.js [hunk 2]
```diff
@@ -5,8 +5,11 @@ import Row from "react-bootstrap/Row";
 import Spinner from "react-bootstrap/Spinner";
 import Card from "react-bootstrap/Card";
 import { HabitatScatterGraph } from "../components/ScatterGraph";
+import { useSelector } from "react-redux";
 
-export default function Habitat({ habitatData }) {
+export default function Habitat() {
+  const {analysisState} = useSelector((state) => state.analysisStateSliceReducer);
+  const habitatData = analysisState["habitat"];
   const habitatIsDemo = habitatData.predictions?.find(
```
[the remark this answers is not in the document]
Comment (johncalesp) [reply deepview-explore/react-ui/src/sections/Habitat.js]: yeah, I refactored that part

**Comment by johncalesp** [comment]
I added more comments in redux/store/store.js to indicate the responsibilities of each slice.
As of now, redux is handling the major states (analysis, epochs, vscode connection), usually the ones that are shared between 2 or more components. That way each individual component can grab what it needs and transform or modify the data locally to render the information, but they can't modify the global state stored in redux.

**Review by michaelshin — APPROVED** [review]

**Pull request merged**